### DISCUSSION TITLE
Evaluate & Implement findings from a careful review of ZPL-C in contrast with what already exists in V's standard library

### DISCUSSION THREAD

**Discussion by ylluminate** [vlang/v#14661] [Ideas]
As per @BenStigsen's suggestion (see: [Discord discussion](https://discord.com/channels/592103645835821068/592842126304477184/973263382713106432)), evaluate the [ZPL C library](https://github.com/zpl-c/zpl) ("*A C99 cross-platform header-only library oriented towards improving the coding experience.*") for targets to implement in and for V that may be missing and/or implemented as lesser-performant algorithms.

**Comment by depperm** [comment]
comparing `math`, v seem to be missing:

- constants
    - 2/3
    - epsilon
    - 1/tau
    - 1/pi
    - tau/2
    - tau/4
    - tau/8
    - sqrt(3)
    - sqrt(5)
- vectors
    - 2 x, y
    - 3 x, y, z
    - 4 x, y, z, w
- matrix
    - 2 x, y
    - 3 x, y, z
    - 4 z, y, z, w
- quat (quaternion?)
- plane
- frustum
- functions
    - angle_diff(radian a, radian b) // to interpolate angles
    - vector math (see above)
    - mat math (see above)
    - quat math (see above)

**Comment by depperm** [comment]
comparing `hash`, v seem to be missing:

- murmur (32/64) https://en.wikipedia.org/wiki/MurmurHash#Algorithm
- adler32 https://en.wikipedia.org/wiki/Adler-32#The_algorithm
- crc64
- ~~base64_encode~~ (in encoding)
- ~~base64_decode~~ (in encoding)

**Comment by JalonSolov** [comment]
`mat (unsure what this stands for)`

Matrix operations.

**Comment by depperm** [comment]
comparing `strings`, v seem to be missing:
- is_alpha
- ~~is_alpha_num~~ `is_alnum`
- ~~is_hex~~ `is_hex_digit`
- ~~has_prefix~~ `starts_with`
- ~~has_suffix~~ `ends_with`

**Comment by JalonSolov** [comment]
V has some of those string functions, just with different names.

* `has_prefix` == `starts_with`
* `has_suffix` == `ends_with`
* etc.

**Comment by JalonSolov** [comment]
It also as a csv parser (reader) in `encoding.csv`
